### PR TITLE
Refactor how we do unlocked file operations

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/basic/tmpfile-util.c [hunk 1]
```diff
@@ -45,6 +47,8 @@ int fopen_temporary(const char *path, FILE **_f, char **_temp_path) {
                 return -errno;
         }
 
+        (void) __fsetlocking(f, FSETLOCKING_BYCALLER);
```
Comment (poettering) [inline src/basic/tmpfile-util.c]: could you add a comment here explaining why this is OK? (i.e. the comment from the commit msg)

Comment (keszybz) [reply src/basic/tmpfile-util.c]: Done.

**Comment by poettering** [comment]
needs rebase!

**Comment by keszybz** [comment]
Rebased.

**Comment by poettering** [comment]
hmm, the failure on the OOM policy test appears unrelated. merging.

**Pull request merged**